M2/Macaulay2/packages/FastLinAlg.m2

```diff
@@ -705,15 +754,19 @@ Rn(ZZ, Ring) := opts -> (n1, R1) -> (
     fullRank := n-r;
     myRand := 0;
     Q := null; C := null; C2 := null; g := null;
-    --    1/0;
 
     possibleMinors := binomial(n, fullRank)*binomial(numberRelations, fullRank);
     minNumberToCutDown := n1+1;
     local numberOfMinorsCompute;
-    if (opts.MaxMinors === null) then (
-        numberOfMinorsCompute = 10*minNumberToCutDown + 8*ceiling(log_1.3(possibleMinors));)
+    if (instance (opts.MaxMinors, Function)) then (
+        numberOfMinorsCompute = opts.MaxMinors(minNumberToCutDown, possibleMinors); )
+        --5*minNumberToCutDown + 8*ceiling(log_1.3(possibleMinors));)
+    else if (instance(opts.MaxMinors, Number)) then (
+        numberOfMinorsCompute = opts.MaxMinors;)
     else (
-        numberOfMinorsCompute = opts.MaxMinors;);
+        --numberOfMinorsCompute = 5*minNumberToCutDown + 8*ceiling(log_1.3(possibleMinors));
+        error "regularInCodimension:  You must pass MaxMinors either a function or a number.";
```

Comment:
Now you don't have a default number (to be used if "null" is there).  Is that intentional?  You could have it.

Reply:
Hi Dan, it is indeed intentional.  The option has a default value (actually, the same as what null used to force).